datatypes/src/collections/feature_collection.rs

```diff
@@ -496,6 +508,99 @@ where
             self.types.clone(),
         ))
     }
+
+    fn rename_columns<S1, S2>(&self, renamings: &[(S1, S2)]) -> Result<Self::Output>
+    where
+        S1: AsRef<str>,
+        S2: AsRef<str>,
+    {
+        let mut rename_map: HashMap<&str, &str> = HashMap::with_capacity(renamings.len());
```

Comment:
maybe also check:
 - no two `from`-names are identical
 - no two `to`-names are identical

Reply:
e633632

Reply:
don't you still need to check the `rename_map`s `values` are unique?

Reply:
aa8e17f